File: backend/Wms.Api/Domain/Entities/Order.cs
using Wms.Api.Domain.Enums;

namespace Wms.Api.Domain.Entities;

public class Order
{
    public int Id { get; set; }
    public required string OrderNumber { get; set; }
    public int WarehouseId { get; set; }
    public required string CustomerName { get; set; }
    public DateTime? ShippedAtUtc { get; set; }
    public bool IsActive { get; set; } = true;
    public DateTime CreatedAtUtc { get; set; } = DateTime.UtcNow;
    public OrderStatus Status { get; set; } = OrderStatus.Draft;
    public Warehouse Warehouse { get; set; } = null!;
    public ICollection<OrderLine> Lines { get; set; } = [];
}

File: frontend/wms-web/src/types/order.ts
export type OrderStatus =
  | 'Draft'
  | 'ReadyToPick'
  | 'Picking'
  | 'Shipping'
  | 'Completed'
  | 'Cancelled'

export type OrderLine = {
  id: number
  productId: number
  productSku: string
  productName: string
  quantity: number
}

export type Order = {
  id: number
  orderNumber: string
  warehouseId: number
  warehouseName: string
  customerName: string
  status: OrderStatus
  isActive: boolean
  createdAtUtc: string
  shippedAtUtc: string | null
  lines: OrderLine[]
}

export type OrderLineRequest = {
  productId: number
  quantity: number
}

export type SaveOrderRequest = {
  warehouseId: number
  customerName: string
  lines: OrderLineRequest[]
}
